aten/src/ATen/native/native_functions.yaml

```diff
@@ -1122,6 +1122,10 @@
     SparseCUDA: bmm_out_sparse_cuda
     SparseCsrCUDA: bmm_out_sparse_csr_cuda
 
+- func: _NestedTensor_GeneralizedBMM(Tensor self, Tensor mat2) -> Tensor
```

Comment:
Let's use the dispatch system to register a kernel for matmul cc @jbschlosser

Reply:
I guess we can't, since matmul is composite...

Reply:
Right, we looked into this already - we need a custom backward and matmul is composite (it'd be cool if we could register backwards per-dispatch-key).